packages/copyable/src/Copyable.tsx

```diff
@@ -0,0 +1,283 @@
+  code: {
+    text: string;
+    background: string;
+    border: string;
+  };
+}
+
+const colorSets: Record<Variant, ColorSet> = {
+  [Variant.Default]: {
+    label: uiColors.gray.dark3,
+    description: uiColors.gray.dark1,
+    code: {
+      text: uiColors.gray.dark3,
+      background: uiColors.gray.light3,
+      border: uiColors.gray.light2,
+    },
+  },
+  [Variant.Dark]: {
+    label: uiColors.gray.light1,
+    description: uiColors.gray.light1,
+    code: {
+      text: uiColors.white,
+      background: uiColors.black,
+      border: uiColors.black,
+    },
+  },
+};
+
+const labelStyle = css`
+  font-size: 14px;
+  font-weight: bold;
+  line-height: 20px;
+`;
+
+const descriptionStyle = css`
+  color: ${uiColors.gray.dark1};
+  font-size: 14px;
+  line-height: 20px;
+  margin-bottom: 6px;
+`;
+
+const containerStyle = css`
+  position: relative;
+  height: 48px;
+  width: 400px;
+  margin: 1px 0;
+  overflow: hidden;
+`;
+
+const copyableContainerStyle = css`
+  height: 36px;
+`;
+
+const codeStyle = css`
+  display: inline-flex;
+  align-items: center;
+
+  height: 100%;
+  width: 100%;
+  padding-left: 12px;
+  border: 1px solid;
+  border-radius: 4px;
+
+  font-size: 14px;
+  overflow: hidden;
+`;
+
+const largeCodeStyle = css`
+  font-size: 16px;
+`;
+
+const buttonWrapperStyle = css`
+  position: absolute;
+  z-index: 1;
+  right: 0;
+  top: 0;
+  height: 100%;
+`;
+
+const buttonStyle = css`
+  height: 100%;
+  border-top-left-radius: 0;
+  border-bottom-left-radius: 0;
+  filter: drop-shadow(2px 0 6px ${transparentize(0.7, uiColors.gray.dark1)});
+`;
+
+const iconStyle = css`
+  padding-right: 6px;
+`;
+
+interface CopyableProps {
+  variant?: Variant;
+  children: string;
+  label: string;
+  description?: string;
+  copyable?: boolean;
+  size?: Size;
+}
+
+export default function Copyable({
+  variant = Variant.Default,
+  children,
+  label,
+  description,
+  copyable = true,
+  size = Size.Default,
+}: CopyableProps) {
+  const colorSet = colorSets[variant];
+
+  const [copied, setCopied] = useState(false);
+  const [showCopyButton, setShowCopyButton] = useState(false);
+  const [buttonRef, setButtonRef] = useState<HTMLButtonElement>();
+
+  useEffect(() => {
+    setShowCopyButton(copyable && ClipboardJS.isSupported());
+  }, [copyable]);
+
+  let copyButton: JSX.Element | undefined;
+
+  if (showCopyButton) {
+    let buttonVariant: ButtonVariant;
+
+    switch (variant) {
+      case Variant.Default:
+        buttonVariant = ButtonVariant.Default;
+        break;
+      case Variant.Dark:
+        buttonVariant = ButtonVariant.Dark;
+        break;
+    }
+
+    const trigger = (
+      <Button
+        ref={setButtonRef}
+        variant={buttonVariant}
+        className={buttonStyle}
+        onClick={() => setCopied(true)}
+      >
+        <CopyIcon size="large" className={iconStyle} />
+        Copy
+      </Button>
+    );
+
+    let tooltipVariant: TooltipVariant;
+
+    switch (variant) {
```

Comment:
This will be very cool when we have a provider from where to pull whether or not the consumer is in darkMode